memiavl/db.go

```diff
@@ -97,13 +100,24 @@ func (db *DB) Commit(changeSets MultiChangeSet) ([]byte, int64, error) {
 			}
 
 			// snapshot rewrite succeeded, catchup and switch
-			// TODO prune the old snapshots
 			if err := result.mtree.CatchupWAL(db.wal); err != nil {
 				return nil, 0, fmt.Errorf("catchup failed: %w", err)
 			}
 			if err := db.reloadMultiTree(result.mtree); err != nil {
 				return nil, 0, fmt.Errorf("switch multitree failed: %w", err)
 			}
+			// prune the old snapshots
+			version := uint32(db.lastCommitInfo.Version)
+			latestSnapshot := snapshotName(version)
```

Comment:
```suggestion
			latestSnapshot := snapshotName(db.snapshot.Version())
```

The latest snapshot should be the loaded snapshot version, rather than the latest version.